Src/FluentAssertions/Common/ExpressionExtensions.cs

```diff
@@ -87,29 +88,51 @@ public static MemberPath GetMemberPath<TDeclaringType, TPropertyType>(
                 case ExpressionType.Call:
                     var methodCallExpression = (MethodCallExpression)node;
 
-                    if (methodCallExpression is not { Method.Name: "get_Item", Arguments: [ConstantExpression argumentExpression] })
+                    if (methodCallExpression is not
+                        { Method.Name: "get_Item", Arguments: [ConstantExpression argumentExpression] })
                     {
                         throw new ArgumentException(GetUnsupportedExpressionMessage(expression.Body), nameof(expression));
                     }
 
                     node = methodCallExpression.Object;
-                    segments.Add($"[{argumentExpression.Value}]");
+                    singlePath = $"[{argumentExpression.Value}].{singlePath}";
+                    break;
+                case ExpressionType.New:
+                    var newExpression = (NewExpression)node;
+
+                    foreach (var member in newExpression.Arguments)
+                    {
+                        var expr = member.ToString();
+                        selectors.Add(expr[expr.IndexOf('.', StringComparison.Ordinal)..]);
+                        declaringTypes.Add(((MemberExpression)member).Member.DeclaringType);
+                    }
+
+                    node = null;
                     break;
 
                 default:
                     throw new ArgumentException(GetUnsupportedExpressionMessage(expression.Body), nameof(expression));
             }
         }
 
-        // If any members were accessed in the expression, the first one found is the last member.
         Type declaringType = declaringTypes.FirstOrDefault() ?? typeof(TDeclaringType);
 
-        IEnumerable<string> reversedSegments = segments.AsEnumerable().Reverse();
-        string segmentPath = string.Join(".", reversedSegments);
+        if (singlePath is null)
+        {
+            return selectors.Select(selector =>
+                GetNewInstance<TDeclaringType>(declaringType, selector));
```

Comment:
🔧 This is dangerous as you're returning a lazy-evulated expression.

Reply:
The method itself returning an `IEnumerable<T>`? Should I change that to `List<T>` too?

Reply:
Either that. Or call `ToList`

Reply:
And please don't resolve comments if you have not addressed the comment yet.

Reply:
I have resolved it? (See the `ConvertAll()` instead if `Select`) 

When I write `Select().ToList()` Rider suggested to use `ConvertAll()`

Reply:
Ah yes, I didn't see the change when I looked at your comment. Maybe I've missed it or maybe you pushed your change _after_ I re-reviewed the code.

Reply:
No, this `ConvertAll` thing I pushed before your second comment and after the first comment 😊